src/bindings/python/src/openvino/runtime/opset13/ops.py

```diff
@@ -222,3 +229,57 @@ def scaled_dot_product_attention(
         "causal": causal,
     }
     return _get_node_factory_opset13().create("ScaledDotProductAttention", inputs, attributes)
+
+
+@nameable_op
+def constant(
+    value: NumericData,
+    dtype: Union[NumericType, Type] = None,
+    name: Optional[str] = None,
+    *,
+    shared_memory: Optional[bool] = False,
+) -> Constant:
+    """Create a Constant node from provided value.
+
+    :param value: One of: array of values or scalar to initialize node with.
+    :param dtype: The data type of provided data.
+                  If dtype does not match, data will be converted.
+                  Note: disables sharing of the memory when convertion occurs.
+    :param name: Optional name for output node.
+    :param shared_memory: keyword-only argument.
+                          If `True`, this Constant's memory is being shared with a host,
+                          that means the responsibility of keeping host memory is
+                          on the side of a user. Any action performed on the host
+                          memory is reflected on this Constant's memory!
+                          If `False`, data is being copied to this Constant.
+                          Requires data to be C_CONTIGUOUS if `True`.
+    :return: The Constant node initialized with provided data.
+    """
+    _value = value
+    _shared_memory = shared_memory
+    _dtype = dtype
+    # Convert scalars to:
+    if isinstance(value, int):
+        log.warning("Converting scalar type of undefined bitwidth to 32-bit integer. Memory sharing is disabled by default.")
+        _value, _shared_memory = np.array(value, dtype=np.int32), False
+    elif isinstance(value, float):
+        log.warning("Converting scalar type of undefined bitwidth to 32-bit float. Memory sharing is disabled by default.")
+        _value, _shared_memory = np.array(value, dtype=np.float32), False
+    elif isinstance(value, bool):
+        log.warning("Converting bool type to numpy bool. Memory sharing is disabled by default.")
+        _value, _shared_memory = np.array(value, dtype=np.bool_), False
+    elif isinstance(value, (np.number, np.bool_)):
+        log.warning(f"Converting numpy scalar of {value.dtype} to corresponding type. Memory sharing is disabled by default.")
+        _value, _shared_memory = np.array(value, dtype=value.dtype), False
+    elif isinstance(value, list):
+        log.warning("Converting list type of undefined bitwidth to 32-bit float array. Memory sharing is disabled by default.")
+        _value, _shared_memory = np.array(value, dtype=np.float32), False
```

Comment:
As I understand for a case like:
`constant([9223372036854775807], dtype=ov.Type.i64)`, `constant([9223372036854775807], dtype=np.int64)` 
 or `constant([1.7976931348623157e+308], dtype=ov.Type.f64)`
the list will be always converted to the list of `float32` by default, and then converted back (within the check L81 `if _dtype != _value.dtype:`) to the list of type provided with `dtype` attribute.

So for the example `constant([9223372036854775807], dtype=ov.Type.i64)` there will be two warnings:
"Converting list type of undefined bitwidth to 32-bit float array. Memory sharing is disabled by default."
"Converting value of np.float32 to np.int64. Memory sharing is disabled by default."

Am I right?
I'm just concerned about possible precision or overflow issues.

What do you think about using the `dtype` attribute as a type for the first conversion to the np.array, and use default type only if `dtype` is not provided? 
Similar mechanism could be applied to all of the scalar cases above to avoid this middle cast.
```
_dtype = None
if dtype is not None:
    _dtype = dtype
if isinstance(dtype, Type):
    _dtype = dtype.to_dtype()
        
...
elif isinstance(value, list):
     if _dtype is None:
         log.warning("Converting list type of undefined bitwidth to 32-bit float array. Memory sharing is disabled by default.")
         _dtype = np.float32
     _value, _shared_memory = np.array(value, dtype=_dtype), False

 ```

Reply:
Adjusted, now implementation tries to convert with numpy first and only casts types when dtype is passed.